feat : introduced new filters to elastic search, creationtime & scientific metadata condition

Junjiequan opened the pull request:
## Description

ElasticSearch integration for SciCat as optional feature.
Disable elasticSearch will fall back to the regular mongoDB query.

## Motivation

MongoDB's aggregation method can yield low search accuracy. Longer queries often result in less relevant outcomes, frustrating users.

In contrast, Elasticsearch boasts a robust search engine. It indexes and stores data flexibly, allowing adaptable search methods based on user preferences. It promptly presents the most relevant information using document relevancy scores, significantly enhancing user experiences.

## Tests included/Docs Updated?

- [x] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [x] New packages used/requires npm install? 
- [ ] Toggle added for new features?

bpedersen2 committed:
Add message for instrument creation problems

Change-Id: I1d91d7ea3a66cdec62d9c667eeebc30d5bd1e30b

Junjiequan committed:
feat: elastic search MVP integration

Junjiequan committed:
feat: fulltext query implementation

Junjiequan committed:
Fixed elastic search dto and added hard filter

Junjiequan committed:
[WIP] Feat: Elastic search integration

Junjiequan committed:
Fix: some clean ups and added api endpoint

Junjiequan committed:
feat: added error handler for ES services

Junjiequan committed:
fix: update local docker compose with elastic search

Junjiequan committed:
fix: fixed prepare:local script

Junjiequan committed:
fix: include elastic search env file for local environment

Junjiequan committed:
trigger code check

Junjiequan committed:
Merge remote-tracking branch 'origin/master' into SWAP-3409-elastic-search-install-and-configure-elastic-sear

Junjiequan committed:
fix: minor cleanup

Junjiequan committed:
fix: fix unit tests

Junjiequan committed:
fix: minor cleanup

Junjiequan committed:
fix: include missing variable into the ENV

Junjiequan committed:
fix: update test setup

Junjiequan committed:
fix: specify env file path for docker compose

Junjiequan committed:
fix: fixed  E2E test

Junjiequan committed:
test

Junjiequan committed:
test

Junjiequan committed:
revert to previous tess yml file

Junjiequan committed:
fix: fix dependency issue

Junjiequan committed:
fix: enable normal search when ES search is disabled

Junjiequan committed:
fix: prevent application crashing from failed ES connection

Junjiequan committed:
fix: minor fix for query filter & some cleanup

Junjiequan committed:
fix: minor code cleanup

bolmsten committed:
Merge branch 'master' into SWAP-3409-elastic-search-install-and-configure-elastic-sear

Junjiequan committed:
Merge branch 'master' into SWAP-3409-elastic-search-install-and-configure-elastic-sear

Junjiequan committed:
Feat: sync elastic search data with mongoDB CRUD operation  (#717)

* feat: update ES data along with mongoDB CRUD actions

* feat: sync ES data with mongoDB on CRUD operation

* fix: minor text change

Junjiequan committed:
Merge branch 'master' into SWAP-3409-elastic-search-install-and-configure-elastic-sear

Junjiequan committed:
fix: minor cleanup

nitrosx committed:
Merge branch 'master' into SWAP-3409-elastic-search-install-and-configure-elastic-sear

bpedersen2 committed:
Use separate eslint/prettier step

and remove posttest in package json.

That way lint errors are much more visible in the test stages output.

Change-Id: Icd0383cecd43277805fa9eea4ce5d221dda11b72

Junjiequan committed:
Merge branch 'master' into SWAP-3409-elastic-search-install-and-configure-elastic-sear

Junjiequan committed:
fix: retry if ES connection fails

Junjiequan committed:
test

Junjiequan committed:
test: run test with elastic search enabled

Junjiequan committed:
fix: included unit testing for ES service & opimized ES folder structure

Junjiequan committed:
fix: remove console logs

Junjiequan committed:
fix: test e2e

Junjiequan committed:
chore: cleanup env & docker compose

Junjiequan committed:
chore: update github action with es container

Junjiequan committed:
chore: run api tests with elastic search

Junjiequan committed:
chore: remove port exposing of ES on github action

Junjiequan committed:
fix: added missing ENV in github action

Junjiequan committed:
fix: add missing ENV for test.yaml

Junjiequan committed:
Merge branch 'master' into SWAP-3409-elastic-search-install-and-configure-elastic-sear

Junjiequan committed:
fix: remove ports exposure of ES in e2e environment

Junjiequan committed:
fix: create and sync dataset collection on application init

Junjiequan committed:
fix: remove data sync from application initialization

Junjiequan committed:
fix: minor cleanup

bpedersen2 committed:
Add prettier check shortcut format:check

Change-Id: Iedba153243f0613361e761067b24a28d89f0536e

bpedersen2 committed:
Fix eslintrc as recommended

Fix rules flagged by `npx eslint-config-prettier `

See https://github.com/prettier/eslint-config-prettier#what-and-why

Change-Id: I3441f316e92358f7f9c1ee0bd8eb0956f608d3a3

Junjiequan committed:
fix: should not search empty space string

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 6.6.0 to 6.7.0

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 6.6.0 to 6.7.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.7.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump mongoose from 7.5.0 to 7.5.2

Bumps [mongoose](https://github.com/Automattic/mongoose) from 7.5.0 to 7.5.2.
- [Release notes](https://github.com/Automattic/mongoose/releases)
- [Changelog](https://github.com/Automattic/mongoose/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Automattic/mongoose/compare/7.5.0...7.5.2)

---
updated-dependencies:
- dependency-name: mongoose
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Junjiequan committed:
Merge branch 'master' into SWAP-3409-elastic-search-install-and-configure-elastic-sear

Junjiequan committed:
fix: minor refactoring

dependabot[bot] committed:
build(deps-dev): bump @types/node-fetch from 2.6.4 to 2.6.5

Bumps [@types/node-fetch](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node-fetch) from 2.6.4 to 2.6.5.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node-fetch)

---
updated-dependencies:
- dependency-name: "@types/node-fetch"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @nestjs/passport from 10.0.1 to 10.0.2

Bumps [@nestjs/passport](https://github.com/nestjs/passport) from 10.0.1 to 10.0.2.
- [Release notes](https://github.com/nestjs/passport/releases)
- [Changelog](https://github.com/nestjs/passport/blob/master/.release-it.json)
- [Commits](https://github.com/nestjs/passport/compare/10.0.1...10.0.2)

---
updated-dependencies:
- dependency-name: "@nestjs/passport"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @nestjs/config from 3.0.1 to 3.1.1

Bumps [@nestjs/config](https://github.com/nestjs/config) from 3.0.1 to 3.1.1.
- [Release notes](https://github.com/nestjs/config/releases)
- [Changelog](https://github.com/nestjs/config/blob/master/.release-it.json)
- [Commits](https://github.com/nestjs/config/compare/3.0.1...3.1.1)

---
updated-dependencies:
- dependency-name: "@nestjs/config"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @user-office-software/duo-message-broker

Bumps [@user-office-software/duo-message-broker](https://github.com/UserOfficeProject/user-office-lib) from 1.5.2 to 1.6.0.
- [Commits](https://github.com/UserOfficeProject/user-office-lib/commits)

---
updated-dependencies:
- dependency-name: "@user-office-software/duo-message-broker"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump uuid and @types/uuid

Bumps [uuid](https://github.com/uuidjs/uuid) and [@types/uuid](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/uuid). These dependencies needed to be updated together.

Updates `uuid` from 9.0.0 to 9.0.1
- [Changelog](https://github.com/uuidjs/uuid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/uuidjs/uuid/compare/v9.0.0...v9.0.1)

Updates `@types/uuid` from 9.0.3 to 9.0.4
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/uuid)

---
updated-dependencies:
- dependency-name: uuid
  dependency-type: direct:production
  update-type: version-update:semver-patch
- dependency-name: "@types/uuid"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump docker/build-push-action from 4 to 5

Bumps [docker/build-push-action](https://github.com/docker/build-push-action) from 4 to 5.
- [Release notes](https://github.com/docker/build-push-action/releases)
- [Commits](https://github.com/docker/build-push-action/compare/v4...v5)

---
updated-dependencies:
- dependency-name: docker/build-push-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/nodemailer from 6.4.9 to 6.4.10

Bumps [@types/nodemailer](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/nodemailer) from 6.4.9 to 6.4.10.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/nodemailer)

---
updated-dependencies:
- dependency-name: "@types/nodemailer"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump docker/metadata-action from 4 to 5

Bumps [docker/metadata-action](https://github.com/docker/metadata-action) from 4 to 5.
- [Release notes](https://github.com/docker/metadata-action/releases)
- [Upgrade guide](https://github.com/docker/metadata-action/blob/master/UPGRADE.md)
- [Commits](https://github.com/docker/metadata-action/compare/v4...v5)

---
updated-dependencies:
- dependency-name: docker/metadata-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Junjiequan committed:
fix: added userGroups check with search query

dependabot[bot] committed:
build(deps-dev): bump sinon from 15.2.0 to 16.0.0

Bumps [sinon](https://github.com/sinonjs/sinon) from 15.2.0 to 16.0.0.
- [Release notes](https://github.com/sinonjs/sinon/releases)
- [Changelog](https://github.com/sinonjs/sinon/blob/main/docs/changelog.md)
- [Commits](https://github.com/sinonjs/sinon/compare/v15.2.0...v16.0.0)

---
updated-dependencies:
- dependency-name: sinon
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Junjiequan committed:
fix: refactor query-builder class

dependabot[bot] committed:
build(deps-dev): bump @types/node from 20.6.1 to 20.6.3

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.6.1 to 20.6.3.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump docker/setup-buildx-action from 2 to 3

Bumps [docker/setup-buildx-action](https://github.com/docker/setup-buildx-action) from 2 to 3.
- [Release notes](https://github.com/docker/setup-buildx-action/releases)
- [Commits](https://github.com/docker/setup-buildx-action/compare/v2...v3)

---
updated-dependencies:
- dependency-name: docker/setup-buildx-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump docker/login-action from 2 to 3

Bumps [docker/login-action](https://github.com/docker/login-action) from 2 to 3.
- [Release notes](https://github.com/docker/login-action/releases)
- [Commits](https://github.com/docker/login-action/compare/v2...v3)

---
updated-dependencies:
- dependency-name: docker/login-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Junjiequan committed:
fix: included userGuards for elastic search controllers

Junjiequan committed:
fix: es-lint test

Junjiequan committed:
fix: lint fix

Junjiequan committed:
fix: fix lint issue

Junjiequan committed:
Merge branch 'master' into SWAP-3409-elastic-search-install-and-configure-elastic-sear

Junjiequan committed:
fix: lint issue fix

dependabot[bot] committed:
build(deps-dev): bump @types/passport-local from 1.0.35 to 1.0.36

Bumps [@types/passport-local](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/passport-local) from 1.0.35 to 1.0.36.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/passport-local)

---
updated-dependencies:
- dependency-name: "@types/passport-local"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/express-session from 1.17.7 to 1.17.8

Bumps [@types/express-session](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express-session) from 1.17.7 to 1.17.8.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/express-session)

---
updated-dependencies:
- dependency-name: "@types/express-session"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @faker-js/faker from 8.0.2 to 8.1.0

Bumps [@faker-js/faker](https://github.com/faker-js/faker) from 8.0.2 to 8.1.0.
- [Release notes](https://github.com/faker-js/faker/releases)
- [Changelog](https://github.com/faker-js/faker/blob/next/CHANGELOG.md)
- [Commits](https://github.com/faker-js/faker/compare/v8.0.2...v8.1.0)

---
updated-dependencies:
- dependency-name: "@faker-js/faker"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/passport-jwt from 3.0.9 to 3.0.10

Bumps [@types/passport-jwt](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/passport-jwt) from 3.0.9 to 3.0.10.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/passport-jwt)

---
updated-dependencies:
- dependency-name: "@types/passport-jwt"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Junjiequan committed:
chore: auto-merge dependabot patch versions

Junjiequan committed:
chore: auto-merge dependabot patch versions

Junjiequan committed:
fix: new action file for auto merge yaml

Junjiequan committed:
update

dependabot[bot] committed:
build(deps-dev): bump eslint from 8.49.0 to 8.50.0

Bumps [eslint](https://github.com/eslint/eslint) from 8.49.0 to 8.50.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.49.0...v8.50.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @nestjs/cli from 10.1.17 to 10.1.18

Bumps [@nestjs/cli](https://github.com/nestjs/nest-cli) from 10.1.17 to 10.1.18.
- [Release notes](https://github.com/nestjs/nest-cli/releases)
- [Changelog](https://github.com/nestjs/nest-cli/blob/master/.release-it.json)
- [Commits](https://github.com/nestjs/nest-cli/compare/10.1.17...10.1.18)

---
updated-dependencies:
- dependency-name: "@nestjs/cli"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/supertest from 2.0.12 to 2.0.13

Bumps [@types/supertest](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/supertest) from 2.0.12 to 2.0.13.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/supertest)

---
updated-dependencies:
- dependency-name: "@types/supertest"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/lodash from 4.14.198 to 4.14.199

Bumps [@types/lodash](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/lodash) from 4.14.198 to 4.14.199.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/lodash)

---
updated-dependencies:
- dependency-name: "@types/lodash"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 6.7.0 to 6.7.3.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.7.3/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 6.7.0 to 6.7.3.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.7.3/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 20.6.3 to 20.7.1

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.6.3 to 20.7.1.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Junjiequan committed:
fix: auto-merge dependabot PR

Junjiequan committed:
fix: removed actions/checkout

Junjiequan committed:
fix: dependabot auto merge

dependabot[bot] committed:
build(deps-dev): bump @types/node-fetch from 2.6.5 to 2.6.6

Bumps [@types/node-fetch](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node-fetch) from 2.6.5 to 2.6.6.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node-fetch)

---
updated-dependencies:
- dependency-name: "@types/node-fetch"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 20.7.1 to 20.8.0

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.7.1 to 20.8.0.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump mongoose from 7.5.2 to 7.5.3

Bumps [mongoose](https://github.com/Automattic/mongoose) from 7.5.2 to 7.5.3.
- [Release notes](https://github.com/Automattic/mongoose/releases)
- [Changelog](https://github.com/Automattic/mongoose/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Automattic/mongoose/compare/7.5.2...7.5.3)

---
updated-dependencies:
- dependency-name: mongoose
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Junjiequan committed:
chore: added rebase auto

dependabot[bot] committed:
build(deps-dev): bump @types/express from 4.17.17 to 4.17.18

Bumps [@types/express](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/express) from 4.17.17 to 4.17.18.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/express)

---
updated-dependencies:
- dependency-name: "@types/express"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @nestjs/swagger from 7.1.11 to 7.1.12

Bumps [@nestjs/swagger](https://github.com/nestjs/swagger) from 7.1.11 to 7.1.12.
- [Release notes](https://github.com/nestjs/swagger/releases)
- [Changelog](https://github.com/nestjs/swagger/blob/master/.release-it.json)
- [Commits](https://github.com/nestjs/swagger/compare/7.1.11...7.1.12)

---
updated-dependencies:
- dependency-name: "@nestjs/swagger"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/nodemailer from 6.4.10 to 6.4.11

Bumps [@types/nodemailer](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/nodemailer) from 6.4.10 to 6.4.11.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/nodemailer)

---
updated-dependencies:
- dependency-name: "@types/nodemailer"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump rimraf from 5.0.1 to 5.0.5

Bumps [rimraf](https://github.com/isaacs/rimraf) from 5.0.1 to 5.0.5.
- [Changelog](https://github.com/isaacs/rimraf/blob/main/CHANGELOG.md)
- [Commits](https://github.com/isaacs/rimraf/compare/v5.0.1...v5.0.5)

---
updated-dependencies:
- dependency-name: rimraf
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/mocha from 10.0.1 to 10.0.2

Bumps [@types/mocha](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/mocha) from 10.0.1 to 10.0.2.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/mocha)

---
updated-dependencies:
- dependency-name: "@types/mocha"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 6.7.0 to 6.7.3

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 6.7.0 to 6.7.3.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.7.3/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 6.7.0 to 6.7.3

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 6.7.0 to 6.7.3.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.7.3/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump chai from 4.3.8 to 4.3.10

Bumps [chai](https://github.com/chaijs/chai) from 4.3.8 to 4.3.10.
- [Release notes](https://github.com/chaijs/chai/releases)
- [Changelog](https://github.com/chaijs/chai/blob/4.x.x/History.md)
- [Commits](https://github.com/chaijs/chai/compare/v4.3.8...v4.3.10)

---
updated-dependencies:
- dependency-name: chai
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump mathjs from 11.11.0 to 11.11.1

Bumps [mathjs](https://github.com/josdejong/mathjs) from 11.11.0 to 11.11.1.
- [Changelog](https://github.com/josdejong/mathjs/blob/develop/HISTORY.md)
- [Commits](https://github.com/josdejong/mathjs/compare/v11.11.0...v11.11.1)

---
updated-dependencies:
- dependency-name: mathjs
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump migrate-mongo from 10.0.0 to 11.0.0

Bumps [migrate-mongo](https://github.com/seppevs/migrate-mongo) from 10.0.0 to 11.0.0.
- [Changelog](https://github.com/seppevs/migrate-mongo/blob/master/CHANGELOG.md)
- [Commits](https://github.com/seppevs/migrate-mongo/commits)

---
updated-dependencies:
- dependency-name: migrate-mongo
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Junjiequan committed:
fix: eslint issue fix

Junjiequan committed:
feat: added date filter & condition filter

Junjiequan committed:
fix: refactor & fixed metadataKeys permission

bpedersen2 commented on CI/E2E/docker-compose-local.yaml:
```diff
@@ -16,12 +16,35 @@ services:
+  es01:
+    depends_on:
+      - mongodb
+    image: docker.elastic.co/elasticsearch/elasticsearch:${STACK_VERSION}
```
would this also work with one of the free(er) clones, e.g. opensearch?

Junjiequan replied:
I haven't looked for openSearch yet, but I think there will be some modification required to replace elastic search with current setup

nitrosx replied:
WE should open a new issue to investigate

dependabot[bot] committed:
build(deps-dev): bump @types/supertest from 2.0.13 to 2.0.14

Bumps [@types/supertest](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/supertest) from 2.0.13 to 2.0.14.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/supertest)

---
updated-dependencies:
- dependency-name: "@types/supertest"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint from 8.50.0 to 8.51.0

Bumps [eslint](https://github.com/eslint/eslint) from 8.50.0 to 8.51.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.50.0...v8.51.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/node from 20.8.0 to 20.8.3

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.8.0 to 20.8.3.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump openid-client from 5.5.0 to 5.6.0

Bumps [openid-client](https://github.com/panva/node-openid-client) from 5.5.0 to 5.6.0.
- [Release notes](https://github.com/panva/node-openid-client/releases)
- [Changelog](https://github.com/panva/node-openid-client/blob/main/CHANGELOG.md)
- [Commits](https://github.com/panva/node-openid-client/compare/v5.5.0...v5.6.0)

---
updated-dependencies:
- dependency-name: openid-client
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump mongoose from 7.5.3 to 7.6.0

Bumps [mongoose](https://github.com/Automattic/mongoose) from 7.5.3 to 7.6.0.
- [Release notes](https://github.com/Automattic/mongoose/releases)
- [Changelog](https://github.com/Automattic/mongoose/blob/master/CHANGELOG.md)
- [Commits](https://github.com/Automattic/mongoose/compare/7.5.3...7.6.0)

---
updated-dependencies:
- dependency-name: mongoose
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 6.7.3 to 6.7.4

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 6.7.3 to 6.7.4.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.7.4/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump nodemailer from 6.9.5 to 6.9.6

Bumps [nodemailer](https://github.com/nodemailer/nodemailer) from 6.9.5 to 6.9.6.
- [Release notes](https://github.com/nodemailer/nodemailer/releases)
- [Changelog](https://github.com/nodemailer/nodemailer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/nodemailer/nodemailer/compare/v6.9.5...v6.9.6)

---
updated-dependencies:
- dependency-name: nodemailer
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump ts-loader from 9.4.4 to 9.5.0

Bumps [ts-loader](https://github.com/TypeStrong/ts-loader) from 9.4.4 to 9.5.0.
- [Release notes](https://github.com/TypeStrong/ts-loader/releases)
- [Changelog](https://github.com/TypeStrong/ts-loader/blob/main/CHANGELOG.md)
- [Commits](https://github.com/TypeStrong/ts-loader/compare/v9.4.4...v9.5.0)

---
updated-dependencies:
- dependency-name: ts-loader
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 6.7.3 to 6.7.4.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v6.7.4/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump sinon from 16.0.0 to 16.1.0

Bumps [sinon](https://github.com/sinonjs/sinon) from 16.0.0 to 16.1.0.
- [Release notes](https://github.com/sinonjs/sinon/releases)
- [Changelog](https://github.com/sinonjs/sinon/blob/main/docs/changelog.md)
- [Commits](https://github.com/sinonjs/sinon/compare/v16.0.0...v16.1.0)

---
updated-dependencies:
- dependency-name: sinon
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Junjiequan committed:
fix: improved dynamic mapping & added test for ES filter.

Junjiequan committed:
fix: test name minor change

Junjiequan committed:
chore: revert to before merging master

Junjiequan committed:
Merge branch 'master' into SWAP-3516-elastic-search-include-filter-for-creationtime

Junjiequan committed:
fix: use original isPublished field query & minor cleanup

Junjiequan committed:
fix: added sharedWith ES query

Junjiequan committed:
fix: use original mode field to apply match : {_id: ids} query

Junjiequan committed:
fix: type error fix with accessGroups

Junjiequan committed:
fix: test

Junjiequan committed:
fix: test fix

Junjiequan committed:
fix: fixed test case wrong value

Junjiequan committed:
fix: minor revert commented code

Junjiequan committed:
fix: added timeout for creating dataset

Junjiequan committed:
fix: fix test case timeout issue

Junjiequan committed:
fix: added wait_for option for inserting ES data insert

Junjiequan committed:
fix: update test yaml with wait_for option

Junjiequan committed:
fix: checkDifferentDatasetTypes test case optimized

Junjiequan committed:
fix: added wait_for options to ES deletion endpoint

Junjiequan committed:
chore: run test with matrix

Junjiequan committed:
chore: conditionally run elastic search test

Junjiequan committed:
chore: disabled fail-fast for API tests

nitrosx committed:
Merge branch 'master' into SWAP-3516-elastic-search-include-filter-for-creationtime
